platform.md

```diff
@@ -7,57 +7,77 @@ A platform orchestrates a lifecycle to make buildpack functionality available to
+      1. [metadata.toml (TOML)](#metadatatoml-toml)
+      1. [order.toml (TOML)](#ordertoml-toml)
+      1. [project-metadata.toml (TOML)](#project-metadatatoml-toml)
+      1. [stack.toml (TOML)](#group.toml-(toml))
+   1. [Labels](#labels)
+      1. [io.buildpacks.build.metadata (JSON)](#iobuildpacksbuildmetadata-json)
+      1. [io.buildpacks.lifecycle.metadata (JSON)](#iobuildpackslifecyclemetadata-json)
+      1. [io.buildpacks.project.metadata (JSON)](#iobuildpacksprojectmetadata-json)
 
 
 ## Platform API Version
 
+**Current Version**: 0.3
+
 The Platform API version:
  - MUST be in form `<major>.<minor>` or `<major>`, where `<major>` is equivalent to `<major>.0`
- - MUST describe the implemented Platform API.
- - SHALL indicate compatibility with a given lifecycle according to the following rules:
-    - When `<major>` is `0`, the platform is only compatible with lifecycles implementing that exact Platform API.
-    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
-    `<major>.<minor>`, where `<major>` of the lifecycle equals `<major>` of the platform and `<minor>` of the lifecycle
-    is greater than or equal to `<minor>` of the platform.
-
-### Compatibility Verification
+ - When `<major>` is greater than `0` increments to `<minor>` SHALL exclusively indicate additive changes
 
-The lifecycle SHALL verify compatibility if the environment variable `CNB_PLATFORM_API` is set. The value of this
-environment variable MUST be the version of the Platform API the platform implements. Compatibility verification SHALL
-NOT occur if this environment variable is not set. Compatibility verification SHALL occur before any other validation.
-
-## Stacks
+## Terminology
 
 A **stack** is a contract defined by a base run OCI image and a base build OCI image that are only updated with security patches.
 Stack images can be modified with mixins in order to make additive changes to the contract.
 
 A **mixin** is a named set of additions to a stack that avoid changing the behavior of buildpacks or apps that do not depend on the mixin.
 
-A **launch layer** refers to a layer in the app OCI image created from a  `<layers>/<layer>` directory as specified in the [Buildpack Interface Specification](buildpack.md).
+An **app image** refers to an OCI image generated by the lifecycle by extending the run image with any or all of the following: app layers, launch layers, launcher layers, image configuration
 
-An **app layer** refers to a layer created from the `<app>` directory as specified in the [Buildpack Interface Specification](buildpack.md).
+A **launch layer** refers to a layer in the app image created from a  `<layers>/<layer>` directory as specified in the [Buildpack Interface Specification](buildpack.md).
 
+An **app layer** refers to a layer in the app image created from the `<app>` directory as specified in the [Buildpack Interface Specification](buildpack.md).
```

Comment:
should this maybe be `<workspace>` directory? idk, i've given up on avoiding the use of "app".

Reply:
We actually call it `<app>` in the buildpack spec and the flags and env vars are consistent with that (`-app` and `CNB_APP_DIR`). The app dir defaults to `/workspace` for :waves-hands: historical reasons.

Reply:
Isn't it for k8s reasons?

Reply:
Do slices count as part of the app layer?

Reply:
It defaults to `/workspace` for tekton reasons (really knative-build at the time the decision was made). Previously in knative-build you couldn't configure the path at which the source was mounted (it was always `/workspace`).

Reply:
A slice layer is _an_ app layer